src/routes/github/configuration/github-configuration-post.ts

```diff
@@ -38,17 +53,11 @@ export const GithubConfigurationPost = async (req: Request, res: Response): Prom
 	req.addLogFields({ gitHubInstallationId });
 	req.log.info("Received add subscription request");
 
-	// Check if the user that posted this has access to the installation ID they're requesting
 	try {
-		const gitHubUserClient = new GitHubUserClient(githubToken, req.log);
-		const gitHubAppClient = new GitHubAppClient(getCloudInstallationId(gitHubInstallationId), req.log);
-
-		const { data: { login } } =  await gitHubUserClient.getUser();
-		const { data: installation } = await gitHubAppClient.getInstallation(gitHubInstallationId);
-
-		if (!await isUserAdminOfOrganization(gitHubUserClient, installation.account.login, login, installation.target_type)) {
```

Comment:
Moved this code block (43 - 51) to a seperate function as its all isolated to the testing of privileges and made error handling much more readable/targeted.